connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -253,20 +266,25 @@ export async function notionUpsertPageActivity(
       lastUpsertedTs: upsertTs,
       skipReason: "body_too_large",
     });
-    return;
+    return { document: newNotionPage, createdOrMoved };
   }
 
   if (parsedPage && parsedPage.hasBody) {
     upsertTs = new Date().getTime();
     const documentId = `notion-${parsedPage.id}`;
+    const parents = await getParents(dataSourceConfig, {
```

Comment:
setting the parents field during the upsert, 3 options, I chose the first one
  	- grab it with getParents, the method to compute the field from connectors DB => going for that, a little load on connectors DB but least worst situation IMHO
	- grab it from core -- unnecessary load on core DB + qdrant for just the parents field IMHO
	- change the upsert to not touch the parents field -- too dirty, since it's not an upsert then, kind of camouflaged logic specific to a field

Reply:
Def the right approach